libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/EHorizonGraphAccessor.kt

```diff
@@ -0,0 +1,27 @@
+package com.mapbox.navigation.core.trip.session
+
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.core.trip.model.eh.EHorizonEdgeMetadata
+
+/**
+ * Electronic horizon graph accessor interface
+*/
+interface EHorizonGraphAccessor {
```

Comment:
Why do we need this interface? Why not expose this as a class with an internal constructor?

Reply:
what's the issue with interfaces?

Reply:
We do not rely on this abstraction, there's no place for providing a different implementation of those interfaces, so why have them in the first place?

Reply:
If I were a user I would prefer to see a `clean` Interface with only public API, 
I see what methods I can use and what data they return. I can get the same info from a class, but I need separate public API and implementation details.

Reply:
By exposing an interface, we theoretically add an option for developers to implement it. Even if it doesn't make practical sense, that's always an option and adding a method to that interface is a breaking change. In my opinion, we should always choose to limit the API surface over obfuscating the source code in the way that's proposed. Interfaces should have an architectural and practical reason to be exposed.

Reply:
we can make `internal` interface to prevent it implementation by users. 
If we change our API it will be a breaking change anyway, no matter we use a class or an interface.

Reply:
If we make it `internal` we won't be able to define it as a return type, so developers would first see the `class` anyway.

Reply:
> If we change our API it will be a breaking change anyway

Adding a method to class is not a breaking change while adding a method to an interface is.

Reply:
ok, you won :)